### PR TITLE
Add confirm password to validator accounts create

### LINKED ISSUE #5715: Confirm password during "validator accounts create"
# 🚀 Feature Request

### Description
"validator accounts create" asks for a password when run without the --password flag, however the prompt is easy to miss, and doesn't provide good UX.  Many users are reporting that they can't decrypt their keys, likely due to stray keystrokes or right click pastes at this point.

### Describe the solution you'd like
Improve the UX for providing interactive password and require confirmation of the provided password to ensure it was entered correctly.

### PR DESCRIPTION
**Pull request by 0xKiwi**
**What type of PR is this?**
Improvement 

**What does this PR do? Why is it needed?**
This PR adds support for `EnterPassword` to optionally ask for a confirmation. If the confirmed password does not align with what the user previously entered, the terminal will ask for a new password to be entered.  

Fixes #5715 

```
[2020-05-06 02:15:07]  INFO accounts: Please enter the password for your private keys
[2020-05-06 02:15:07]  INFO node: Enter a password:
[2020-05-06 02:15:10]  INFO node: Please enter your password again:
[2020-05-06 02:15:12]  INFO node: Passwords did not match, please try again
[2020-05-06 02:15:12]  INFO node: Enter a password:
```

### CODE REVIEW

File: shared/cmd/helpers.go [hunk 1]
```diff
@@ -57,5 +58,18 @@ func EnterPassword() (string, error) {
 	}
 	text := string(bytePassword)
 	passphrase = strings.Replace(text, "\n", "", -1)
+	if confirmPassword {
+		log.Info("Please enter your password again:")
```
Comment (shayzluf) [inline shared/cmd/helpers.go:62]:
```suggestion
		log.Info("Please re-enter your password:")
```
maybe better

File: shared/cmd/helpers.go [hunk 2]
```diff
@@ -57,5 +58,18 @@ func EnterPassword() (string, error) {
...
+		bytePassword, err := terminal.ReadPassword(int(os.Stdin.Fd()))
+		if err != nil {
+			return passphrase, errors.Wrap(err, "could not read account password")
```
Comment (shayzluf) [inline shared/cmd/helpers.go:65]:
```suggestion
			return nil, errors.Wrap(err, "could not read account password")
```

**Comment by shayzluf** [comment]
maybe it will be good to test the reenter path as well

File: shared/cmd/helpers.go [hunk 3]
```diff
@@ -3,10 +3,11 @@ package cmd
 import (
 	"bufio"
 	"fmt"
-	"github.com/pkg/errors"
 	"os"
 	"strings"
 
+	"github.com/pkg/errors"
+
```
Comment (terencechain) [inline shared/cmd/helpers.go:10]:
```suggestion
```

File: shared/cmd/helpers.go [hunk 4]
```diff
@@ -48,7 +49,7 @@ func ConfirmAction(actionText string, deniedText string) (bool, error) {
 // EnterPassword queries the user for their password through the terminal, in order to make sure it is
 // not passed in a visible way to the terminal.
 // TODO(#5749): This function is untested and should be tested.
-func EnterPassword() (string, error) {
+func EnterPassword(confirmPassword bool) (string, error) {
```
Comment (terencechain) [inline shared/cmd/helpers.go:52]: We really should just make `confirmPassword` as default. We shouldn't even be using the non-confirm password mode

Comment (0xKiwi) [reply shared/cmd/helpers.go:51]: Even for standard validator start up?

Comment (terencechain) [reply shared/cmd/helpers.go:51]: Didnt know this is used for standard start up. I think it's fine then

Comment (0xKiwi) [reply shared/cmd/helpers.go:51]: Its also used for `accounts keys`, but no password confirmation is required there.

**Pull request merged**